src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SelectedMembers/AbstractSelectedMembers.cs

```diff
@@ -28,13 +29,19 @@ internal abstract class AbstractSelectedMembers<
         where TVariableSyntax : SyntaxNode
     {
         protected abstract SyntaxList<TMemberDeclarationSyntax> GetMembers(TTypeDeclarationSyntax containingType);
-        protected abstract IEnumerable<TVariableSyntax> GetAllDeclarators(TFieldDeclarationSyntax field);
-
-        protected abstract SyntaxToken GetVariableIdentifier(TVariableSyntax declarator);
-        protected abstract SyntaxToken GetPropertyIdentifier(TPropertyDeclarationSyntax declarator);
+        protected abstract IEnumerable<(SyntaxToken identifier, SyntaxNode declaration)> GetDeclaratorsAndIdentifiers(TMemberDeclarationSyntax member);
 
         public async Task<ImmutableArray<SyntaxNode>> GetSelectedFieldsAndPropertiesAsync(
             SyntaxTree tree, TextSpan textSpan, bool allowPartialSelection, CancellationToken cancellationToken)
+                => await GetSelectedMembersAsync(tree, textSpan, allowPartialSelection, IsFieldOrProperty, cancellationToken).ConfigureAwait(false);
```

Comment:
Refactor the current logic here to use a common method, which can take a filter if we only want certain members